### ISSUE TITLE
Server ticking entity when using Thaumcraft Golems and Ars Magic 2.5 Version 1.6.1 and 1.6.2. Boots the player off

### ISSUE THREAD

**Issue by GraysonUnitt** [TCLProject/ArsMagica2-5#12]
Unfortunately when a player is near a Thaumcraft golem with AM 2.5 version 1.6.1 or 1.6.2 on a server it will boot the player off saying there is a no such method error and then a ticking player exception occurs. Tested with a fresh world on the server as well. This was the exception I was able to grab from server logs :
```
[20:06:32] [Server thread/ERROR] [FML]: Exception caught during firing event cpw.mods.fml.common.gameevent.TickEvent$PlayerTickEvent@6dec697a:
java.lang.NoSuchMethodError: net.tclproject.mysteriumlib.asm.fixes.MysteriumPatchesFixesMagicka.distanceTo(Lnet/minecraft/util/Vec3;Lnet/minecraft/util/Vec3;)D
        at net.minecraft.util.Vec3.func_72438_d(Vec3.java) ~[azw.class:?]
        at chylex.hee.mechanics.compendium.events.CompendiumEvents.getObservation(CompendiumEvents.java:89) ~[CompendiumEvents.class:?]
        at chylex.hee.mechanics.compendium.events.CompendiumEvents.onPlayerTick(CompendiumEvents.java:142) ~[CompendiumEvents.class:?]
        at cpw.mods.fml.common.eventhandler.ASMEventHandler_346_CompendiumEvents_onPlayerTick_PlayerTickEvent.invoke(.dynamic) ~[?:?]
        at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) ~[ASMEventHandler.class:?]
        at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140) [EventBus.class:?]
        at cpw.mods.fml.common.FMLCommonHandler.onPlayerPreTick(FMLCommonHandler.java:345) [FMLCommonHandler.class:?]
        at net.minecraft.entity.player.EntityPlayer.func_70071_h_(Unknown Source) [yz.class:?]
        at net.minecraft.entity.player.EntityPlayerMP.func_71127_g(EntityPlayerMP.java:295) [mw.class:?]
        at net.minecraft.network.NetHandlerPlayServer.func_147347_a(NetHandlerPlayServer.java:303) [nh.class:?]
        at net.minecraft.network.play.client.C03PacketPlayer.func_148833_a(SourceFile:137) [jd.class:?]
        at net.minecraft.network.play.client.C03PacketPlayer$C05PacketPlayerLook.func_148833_a(SourceFile:98) [jg.class:?]
        at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212) [ej.class:?]
        at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:165) [nc.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:659) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:334) [lt.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:427) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685) [li.class:?]
[20:06:32] [Server thread/ERROR] [FML]: Index: 5 Listeners:
[20:06:32] [Server thread/ERROR] [FML]: 0: NORMAL
[20:06:32] [Server thread/ERROR] [FML]: 1: ASM: cofh.mod.updater.UpdateManager@af14fe7 tickStart(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 2: ASM: mods.railcraft.common.items.firestone.FirestoneTickHandler@15874d4d tick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 3: ASM: cofh.mod.updater.UpdateManager@53c1f522 tickStart(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 4: ASM: mods.battlegear2.BattlegearTickHandeler@3fc57eff onPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 5: ASM: chylex.hee.mechanics.compendium.events.CompendiumEvents@49c762f5 onPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 6: ASM: cofh.mod.updater.UpdateManager@12ce81d4 tickStart(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 7: ASM: tconstruct.armor.ArmorAbilities@3ee3ebc5 playerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 8: ASM: com.emoniph.witchery.common.ServerTickEvents@2f170a11 onPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 9: ASM: emt.util.EMTEventHandler@43fe5410 onPlayerTickEvent(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 10: ASM: cofh.mod.updater.UpdateManager@40e18c2c tickStart(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 11: ASM: witchinggadgets.common.util.handler.PlayerTickHandler@21dc56ea playerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 12: ASM: net.tclproject.immersivecavegen.ImmersiveCavegen@777337f5 playerBlind(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 13: ASM: tschallacka.magiccookies.eventhandlers.EventHandlerWorld@3fc33f0d onPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 14: ASM: codechicken.nei.ServerHandler@6aab0dd4 tickEvent(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 15: ASM: alexsocol.patcher.handler.PatcherEventHandler@471d57af onPlayerPreTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 16: ASM: thaumic.tinkerer.common.potion.PotionEffectHandler@2b883514 onPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 17: ASM: alfheim.common.core.handler.EventHandler@66e79967 onPlayerUpdate(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 18: ASM: alfheim.common.core.handler.ESMHandler@1846e510 onPlayerUpdate(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 19: ASM: alfheim.common.core.handler.ElvenFlightHandler@24e5c23b onPlayerUpdate(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 20: ASM: alfheim.common.core.helper.ContributorsPrivacyHelper@22ac85af onPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 21: ASM: mods.railcraft.common.plugins.forestry.IceManTickHandler@4c656893 tick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 22: ASM: travellersgear.common.util.TGEventHandler@244f946a playerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 23: ASM: makamys.coretweaks.bugfix.DoubleEatFixer@57a4d92b onClientTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 24: ASM: ganymedes01.etfuturum.core.handlers.ServerEventHandler@6df2326 postPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 25: ASM: net.dries007.holoInventory.server.ServerEventHandler@176cb3ec event(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 26: ASM: com.denfop.events.EventUpdate@4527c746 tickStart(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 27: ASM: mekanism.common.CommonPlayerTickHandler@530c1238 onTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 28: ASM: DelirusCrux.Netherlicious.Achievements.BiomeAchievements@5bb5a7e6 biomeAchievements(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 29: ASM: tombenpotter.sanguimancy.util.EventHandler@745cee85 onPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 30: ASM: am2.armor.ArmorEventHandler@476021eb onPlayerTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 31: ASM: vazkii.botania.common.block.BlockPistonRelay@74f7d737 tickEnd(Lcpw/mods/fml/common/gameevent/TickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 32: ASM: lumien.randomthings.Handler.RTTickHandler@3254125b tick(Lcpw/mods/fml/common/gameevent/TickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 33: ASM: gollorum.signpost.SPEventHandler@6b4bbd6e onClientTick(Lcpw/mods/fml/common/gameevent/TickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 34: ASM: gollorum.signpost.SPEventHandler@6b4bbd6e onServerTick(Lcpw/mods/fml/common/gameevent/TickEvent;)V
[20:06:32] [Server thread/ERROR] [FML]: 35: LOWEST
[20:06:32] [Server thread/ERROR] [FML]: 36: ASM: vazkii.botania.common.item.ItemRegenIvy@32b0872d onTick(Lcpw/mods/fml/common/gameevent/TickEvent$PlayerTickEvent;)V
[20:06:32] [Server thread/WARN]: Failed to handle packet for /100.0.217.213:52438
net.minecraft.util.ReportedException: Ticking player
        at net.minecraft.entity.player.EntityPlayerMP.func_71127_g(EntityPlayerMP.java:349) ~[mw.class:?]
        at net.minecraft.network.NetHandlerPlayServer.func_147347_a(NetHandlerPlayServer.java:303) ~[nh.class:?]
        at net.minecraft.network.play.client.C03PacketPlayer.func_148833_a(SourceFile:137) ~[jd.class:?]
        at net.minecraft.network.play.client.C03PacketPlayer$C05PacketPlayerLook.func_148833_a(SourceFile:98) ~[jg.class:?]
        at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212) ~[ej.class:?]
        at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:165) [nc.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:659) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:334) [lt.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:427) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685) [li.class:?]
Caused by: java.lang.NoSuchMethodError: net.tclproject.mysteriumlib.asm.fixes.MysteriumPatchesFixesMagicka.distanceTo(Lnet/minecraft/util/Vec3;Lnet/minecraft/util/Vec3;)D
        at net.minecraft.util.Vec3.func_72438_d(Vec3.java) ~[azw.class:?]
        at chylex.hee.mechanics.compendium.events.CompendiumEvents.getObservation(CompendiumEvents.java:89) ~[CompendiumEvents.class:?]
        at chylex.hee.mechanics.compendium.events.CompendiumEvents.onPlayerTick(CompendiumEvents.java:142) ~[CompendiumEvents.class:?]
        at cpw.mods.fml.common.eventhandler.ASMEventHandler_346_CompendiumEvents_onPlayerTick_PlayerTickEvent.invoke(.dynamic) ~[?:?]
        at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) ~[ASMEventHandler.class:?]
        at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140) ~[EventBus.class:?]
        at cpw.mods.fml.common.FMLCommonHandler.onPlayerPreTick(FMLCommonHandler.java:345) ~[FMLCommonHandler.class:?]
        at net.minecraft.entity.player.EntityPlayer.func_70071_h_(Unknown Source) ~[yz.class:?]
        at net.minecraft.entity.player.EntityPlayerMP.func_71127_g(EntityPlayerMP.java:295) ~[mw.class:?]
        ... 10 more
[20:06:32] [Server thread/INFO]: Johnmerlin lost connection: TextComponent{text='Internal server error', siblings=[], style=Style{hasParent=false, color=null, bold=null, italic=null, underlined=null, obfuscated=null, clickEvent=null, hoverEvent=null}}
```

Works completely fine in single player. I was able to revert back to version 1.5.0 hotfix and confirm the issue does not exist in that version.

**Comment by TCLProject** [comment]
Potentially fixed in 1.6.3. Test it and see if the issue still occurs, and come back to me.

**Comment by TCLProject** [comment]
Confirmed fixed.
